pymeasure/instruments/ametek/ametek7270.py

```diff
@@ -50,161 +71,244 @@ class Ametek7270(Instrument):
         """ A floating point property that represents the output
         value on DAC2 in Volts. This property can be set. """,
         validator=truncated_range,
-        values=[-10, 10]
+        values=[-10, 10],
+        check_set_errors=True,
     )
     dac3 = Instrument.control(
         "DAC. 3", "DAC. 3 %g",
         """ A floating point property that represents the output
         value on DAC3 in Volts. This property can be set. """,
         validator=truncated_range,
-        values=[-10, 10]
+        values=[-10, 10],
+        check_set_errors=True,
     )
     dac4 = Instrument.control(
         "DAC. 4", "DAC. 4 %g",
         """ A floating point property that represents the output
         value on DAC4 in Volts. This property can be set. """,
         validator=truncated_range,
-        values=[-10, 10]
+        values=[-10, 10],
+        check_set_errors=True,
     )
     adc1 = Instrument.measurement("ADC. 1",
-                                  """ Reads the input value of ADC1 in Volts """
+                                  """ Reads the input value of ADC1 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     adc2 = Instrument.measurement("ADC. 2",
-                                  """ Reads the input value of ADC2 in Volts """
+                                  """ Reads the input value of ADC2 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     adc3 = Instrument.measurement("ADC. 3",
-                                  """ Reads the input value of ADC3 in Volts """
+                                  """ Reads the input value of ADC3 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     adc4 = Instrument.measurement("ADC. 4",
-                                  """ Reads the input value of ADC4 in Volts """
+                                  """ Reads the input value of ADC4 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     id = Instrument.measurement("ID",
-                                """ Reads the instrument identification """
+                                """ Reads the instrument identification """,
+                                cast=str,
                                 )
 
-    def __init__(self, adapter, name="Ametek DSP 7270", **kwargs):
+    def __init__(self, adapter, name="Ametek DSP 7270",
+                 read_termination='\x00',
+                 write_termination='\x00',
+                 **kwargs):
+
         super().__init__(
             adapter,
             name,
-            **kwargs
-        )
+            read_termination=read_termination,
+            write_termination=write_termination,
+            **kwargs)
+
+    def check_errors(self):
+        """mandatory to be used for property setter

```

Comment:
As it is only for the setter, it should be `check_set_errors` (since #883 ):
```suggestion
    def check_set_errors(self):
        """mandatory to be used for property setter
```